pkgs/development/python-modules/docx2python/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pytestCheckHook }:
+
+buildPythonPackage rec {
+  pname = "docx2python";
+  version = "unstable-2020-11-15";
+
+  # Pypi does not contain tests
+  src = fetchFromGitHub {
+    owner = "ShayHill";
+    repo = pname;
+    rev = "21b2edafc0a01a6cfb73aefc61747a65917e2cad";
+    sha256 = "1nwg17ziwm9a2x7yxsscj8zgc1d383ifsk5w7qa2fws6gf627kyi";
+  };
+
+  preCheck = "cd test"; # Tests require the `test/resources` folder to be accessible
+  checkInputs = [ pytestCheckHook ];
+  disabledTests = [ "test_docx2python.py" "test_docx_context.py" "test_google_docs.py" ]; # Failing tests
```

Comment:
it's fine to disable tests, but do you mind commenting on *why* they need to be disabled?

Reply:
Multiple reasons i do not quite understand:
```
=================================== FAILURES ===================================
______________________ TestImageDir.test_pull_image_files ______________________

self = <test_docx2python.TestImageDir object at 0x7ffff5f3a4c0>

    def test_pull_image_files(self) -> None:
        """Copy image files to output path."""
        docx2python("resources/example.docx", "delete_this/path/to/images")
>       assert os.listdir("delete_this/path/to/images") == ["image1.png", "image2.jpg"]
E       AssertionError: assert ['image2.jpg', 'image1.png'] == ['image1.png', 'image2.jpg']

test_docx2python.py:161: AssertionError
___________________ TestPullImageFiles.test_pull_image_files ___________________

self = <test_docx_context.TestPullImageFiles object at 0x7ffff68868e0>

    def test_pull_image_files(self) -> None:
        """Copy image files to output path."""
        zipf = zipfile.ZipFile("resources/example.docx")
        context = get_context(zipf)
        pull_image_files(zipf, context, "delete_this/path/to/images")
>       assert os.listdir("delete_this/path/to/images") == ["image1.png", "image2.jpg"]
E       AssertionError: assert ['image2.jpg', 'image1.png'] == ['image1.png', 'image2.jpg']

test_docx_context.py:106: AssertionError
____________________ TestPullImageFiles.test_no_image_files ____________________

self = <test_docx_context.TestPullImageFiles object at 0x7ffff5f3a580>

    def test_no_image_files(self) -> None:
        """Pass silently when no image files."""
        zipf = zipfile.ZipFile("resources/basic.docx")
        context = get_context(zipf)
        pull_image_files(zipf, context, "delete_this/path/to/images")
>       assert os.listdir("delete_this/path/to/images") == []
E       AssertionError: assert ['image2.jpg', 'image1.png'] == []

test_docx_context.py:115: AssertionError
_______ TestGoogleDocs.test_empty_properties_dict_if_docProps_not_found ________

self = <test_google_docs.TestGoogleDocs object at 0x7ffff68db880>

    def test_empty_properties_dict_if_docProps_not_found(self) -> None:
        """
        It seems Google Docs docx files to not contain a document properties file:
        `docProps/core.xml`. The contents of this file are returned as a dictionary.
        To correct the above error, result.properties will now return an empty
        dictionary.
        """
>       result = docx2python(TEST_FILE)

test_google_docs.py:56:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/nwjxqi812nsk6k35xjy8h4l7kvc8hcml-python3.8-docx2python-unstable-2020-11-15/lib/python3.8/site-packages/docx2python/main.py:34: in docx2python
    zipf = zipfile.ZipFile(docx_filename)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <zipfile.ZipFile [closed]>
file = '/build/source/test/test_google_docs.py/../resources/test-docx2python-conversion-google_docs.docx'
mode = 'r', compression = 0, allowZip64 = True, compresslevel = None

    def __init__(self, file, mode="r", compression=ZIP_STORED, allowZip64=True,
                 compresslevel=None, *, strict_timestamps=True):
        """Open the ZIP file with mode read 'r', write 'w', exclusive create 'x',
        or append 'a'."""
        if mode not in ('r', 'w', 'x', 'a'):
            raise ValueError("ZipFile requires mode 'r', 'w', 'x', or 'a'")

        _check_compression(compression)

        self._allowZip64 = allowZip64
        self._didModify = False
        self.debug = 0  # Level of printing: 0 through 3
        self.NameToInfo = {}    # Find file info given name
        self.filelist = []      # List of ZipInfo instances for archive
        self.compression = compression  # Method of compression
        self.compresslevel = compresslevel
        self.mode = mode
        self.pwd = None
        self._comment = b''
        self._strict_timestamps = strict_timestamps

        # Check if we were passed a file-like object
        if isinstance(file, os.PathLike):
            file = os.fspath(file)
        if isinstance(file, str):
            # No, it's a filename
            self._filePassed = 0
            self.filename = file
            modeDict = {'r' : 'rb', 'w': 'w+b', 'x': 'x+b', 'a' : 'r+b',
                        'r+b': 'w+b', 'w+b': 'wb', 'x+b': 'xb'}
            filemode = modeDict[mode]
            while True:
                try:
>                   self.fp = io.open(file, filemode)
E                   NotADirectoryError: [Errno 20] Not a directory: '/build/source/test/test_google_docs.py/../resources/test-docx2python-conversion-google_docs.docx'

/nix/store/v3bj7jrns4sk6yj2rp30p6v2l7p707az-python3-3.8.8/lib/python3.8/zipfile.py:1251: NotADirectoryError
=========================== short test summary info ============================
FAILED test_docx2python.py::TestImageDir::test_pull_image_files - AssertionEr...
FAILED test_docx_context.py::TestPullImageFiles::test_pull_image_files - Asse...
FAILED test_docx_context.py::TestPullImageFiles::test_no_image_files - Assert...
FAILED test_google_docs.py::TestGoogleDocs::test_empty_properties_dict_if_docProps_not_found
========================= 4 failed, 81 passed in 3.67s =========================
builder for '/nix/store/rzgq0is0ciy05anzir2mwkwkyjnqc2hq-python3.8-docx2python-unstable-2020-11-15.drv' failed with exit code 1
error: build of '/nix/store/rzgq0is0ciy05anzir2mwkwkyjnqc2hq-python3.8-docx2python-unstable-2020-11-15.drv' failed
```

Reply:
```suggestion
  disabledTests = [
  	# asserts related to file deletions fail
    "test_docx2python.py"
    "test_docx_context.py"
    "test_google_docs.py"
  ];
```

Reply:
Done :)